tomcat-8.5/src/org/dogtagpki/tomcat/JSSContext.java

```diff
@@ -0,0 +1,102 @@
+package org.dogtagpki.tomcat;
+
+import java.util.List;
+import java.security.Provider;
+import java.security.Security;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+
+import org.apache.tomcat.util.net.SSLContext;
+
+import org.mozilla.jss.provider.javax.crypto.JSSKeyManager;
+import org.mozilla.jss.ssl.javax.JSSEngine;
+import org.mozilla.jss.ssl.javax.JSSParameters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JSSContext implements org.apache.tomcat.util.net.SSLContext {
+    public static Logger logger = LoggerFactory.getLogger(JSSContext.class);
+
+    private javax.net.ssl.SSLContext ctx;
+    private JSSParameters params;
+
+    public JSSContext(String alias, List<String> negotiableProtocols) {
+        logger.debug("TomcatJSSContext(" + alias + ", ...");
```

Comment:
The class name in the log messages (TomcatJSSContext) doesn't match the actual class name (JSSContext).

Reply:
(longer explanation than necessary)

So the issue with TomcatJSS is that we've had a history of prefixing all of our classes well, "JSS". `JSSContext` is unique in that it exists in both TomcatJSS and JSS, but implementing/inheriting from different classes.

We have two ways of solving this:

1. (conditionally?) make JSS depend on Tomcat and make `org.mozilla.jss.provider.javax.net.JSSContext` extend `java.security.SSLContextSpi` and implement `org.apache.tomcat.util.net.SSLContext`.
2. Have two separate classes and use the JSS class to implement this class.
   2a. Use the TomcatJSS prefix on the TomcatJSS class.
   2b. Use the same class name in different libraries.

The last (`2b`) is the route I ended up taking. Since a lot of these calls will first call into TomcatJSS and then immediately into JSS (with a class of the same name) I decided to prefix our logging calls with TomcatJSSContext instead of JSSContext to make the difference obvious on a quick read. The log does show the full class name before our log message, so perhaps it is unnecessary. 

Thoughts?